.github/workflows/deploy-api-docs.yml

```diff
@@ -3,55 +3,82 @@ on:
-      with:
-        name: vapor-api-docs
-        template: stack.yaml
-        no-fail-on-empty-changeset: '1'
-        parameter-overrides: >-
-          BucketName=vapor-api-docs-site,
-          SubDomainName=api,
-          HostedZoneName=vapor.codes,
-          AcmCertificateArn=${{ secrets.API_DOCS_CERTIFICATE_ARN }}
-      if: steps.cred.outcome == 'success'
-    - name: Deploy to S3
-      id: s3deploy
-      uses: jakejarvis/s3-sync-action@master
-      with:
-        args: --acl public-read --follow-symlinks
-      env:
-        AWS_S3_BUCKET: 'vapor-api-docs-site'
-        AWS_ACCESS_KEY_ID: ${{ secrets.API_DOCS_DEPLOYER_AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.API_DOCS_DEPLOYER_AWS_SECRET_ACCESS_KEY }}
-        AWS_REGION: 'eu-west-2'
-        SOURCE_DIR: 'public'
-      if: steps.clouddeploy.outcome == 'success'
-    - name: Invalidate CloudFront
-      uses: awact/cloudfront-action@master
-      env:
-        SOURCE_PATH: '/*'
-        AWS_ACCESS_KEY_ID: ${{ secrets.API_DOCS_DEPLOYER_AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.API_DOCS_DEPLOYER_AWS_SECRET_ACCESS_KEY }}
-        AWS_REGION: 'eu-west-2'
-        DISTRIBUTION_ID: ${{ secrets.VAPOR_API_DOCS_DISTRIBUTION_ID }}
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build site
+        run: swift generate-api-docs.swift
+      - name: Install curl and awscliv2
+        run: | 
+          apt-get update && apt-get install -y curl
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          unzip awscliv2.zip
+          ./aws/install
+      - name: Regenerate all API docs if requested
+        if: inputs.regenerate_all
+        shell: bash
+        run: |
+          curl -sL \
+            https://raw.githubusercontent.com/vapor/api-docs/main/generate-package-api-docs.swift -o generate-package-api-docs.swift \
+            https://raw.githubusercontent.com/vapor/api-docs/main/theme-settings.json -o theme-settings.json            
+          for item in \
+            vapor:Vapor vapor:XCTVapor \
+            async-kit:AsyncKit routing-kit:RoutingKit console-kit:ConsoleKit \
+            websocket-kit:WebSocketKit multipart-kit:MultipartKit \
+            postgres-nio:PostgresNIO mysql-nio:MySQLNIO sqlite-nio:SQLiteNIO sql-kit:SQLKit \
+            postgres-kit:PostgresKit mysql-kit:MySQLKit sqlite-kit:SQLiteKit fluent-kit:FluentKit \
+            fluent-kit:FluentSQL fluent-kit:XCTFluent fluent:Fluent fluent-postgres-driver:FluentPostgresDriver \
+            fluent-mongo-driver:FluentMongoDriver fluent-mysql-driver:FluentMySQLDriver fluent-sqlite-driver:FluentSQLiteDriver \
+            redis:Redis queues-redis-driver:QueuesRedisDriver queues:Queues queues:XCTQueues \
+            leaf-kit:LeafKit leaf:Leaf \
+            jwt-kit:JWTKit jwt:JWT apns:APNS \
+          do
+            repo=${item%:*} module=${item#*:}
+            if [[ ! -d ${repo} ]]; then
```

Comment:
Let's merge this but I'm still not 100% sold on this - these 2 lines especially. I have no idea what they do from scanning it and the regex pattern(?) is something I still can't decipher even after a couple of minutes on Google

Reply:
Nope, not a regex pattern - that'd be `${item/:.*//}` and `${item/.*://}` (and even then technically that should be `${item/#:.*//}` and `${item/%.*://}`). It's a Bash syntactic sleight of hand, per `bash(1)`:

>       `${parameter#word}`
>       `${parameter##word}`
>              **`Remove matching prefix pattern.`** `The word is expanded to produce a pattern just as in pathname expansion, and matched against the expanded value of parameter using the rules described under Pattern  Matching below. If the pattern matches the beginning of the value of parameter, then the result of the expansion is the expanded value of parameter with the shortest matching pattern (the ‘‘#’’ case) or the longest matching pattern (the ‘‘##’’ case) deleted. If parameter is @ or *, the pattern removal operation is applied to each positional parameter in turn, and the expansion is the resultant list. If parameter is an array variable subscripted with @ or *, the pattern removal operation is applied to each member of the array in turn, and the expansion is the resultant list.`
>
>       `${parameter%word}`
>       `${parameter%%word}`
>              **`Remove  matching  suffix  pattern.`** `The  word is expanded to produce a pattern just as in pathname expansion, and matched against the expanded value of parameter using the rules described under Pattern Matching below. If the pattern matches a trailing portion of the expanded value of parameter, then the result of the expansion is the expanded value of parameter with the shortest matching pattern (the ‘‘%’’ case) or the longest matching pattern (the  ‘‘%%’’  case) deleted. If parameter is @ or *, the pattern removal operation is applied to each positional parameter in turn, and the expansion is the resultant list.  If parameter is an array variable subscripted with @ or *, the pattern removal operation is applied to each member of the array in turn, and the expansion is the resultant list.`

But you're right, this is utterly inscrutable, I'll yank it.